ui/src/app/google-analytics-events.service.ts

```diff
@@ -0,0 +1,19 @@
+import {Injectable} from "@angular/core";
+declare let gtag: Function;
+ 
+@Injectable()
+export class GoogleAnalyticsEventsService {
+ 
+  public emitEvent(eventName: string,
+		   eventCategory: string,
+                   eventAction: string,
+                   eventLabel: string = null,
+                   eventValue: number = null) {
+    gtag('event', eventName, {
+      eventCategory: eventCategory,
```

Comment:
When writing an object literal, if the key and value share the same name they can be collapsed: ie

```
const eventLabel = 'some label';
const obj = {eventLabel};  // equivalent to obj = {eventLabel: eventLabel};
```

So this becomes shortened to

```
{ eventCategory, eventLabel, eventAction, eventValue }
```

Reply:
Done.